tests/test_views.py

```diff
@@ -0,0 +1,174 @@
+    """
+    Given an iterable of (schema, datetime, user_id, event_type),
+    return a list of tuples where the datetime is formatted as
+    'YYYY-MM-DD HH:MM:SS' plus '+00' only for 'ltt' rows.
+    """
+    out = []
+    for schema, dt, user, ev in rows:
+        ts = dt.strftime('%Y-%m-%d %H:%M:%S')
+        if schema != 'legacy':
+            ts += '+00'
+        out.append((schema, ts, user, ev))
+    return out
+
+def run_test(mock_data, expected_rows):
+    """Wipe, insert mock_data, run view, normalize & assert against expected_rows."""
+    with psycopg2.connect(DATABASE_URL) as conn, conn.cursor() as cur:
+        insert_mock_data(cur, mock_data)
+        conn.commit()
+        raw = query_events_view(cur)
+    actual = normalize_results(raw)
+    assert actual == expected_rows
+
+@pytest.fixture
+def isacc_data():
+    path = os.path.join(DATA_DIR, "isacc_sample.json")
+    with open(path, 'r') as f:
+        return json.load(f)
+
+@pytest.fixture
+def isacc_expected():
+    return [
+        (
+            "3.0",                             # version → schema_version
+            "2025-04-23 14:22:10+00",          # occurred_at
+            "Practitioner/ABC-123",            # user_id (from subject)
+            "login"                            # event_type
+        )
+    ]
+
+@pytest.fixture
+def dhair2_data():
+    path = os.path.join(DATA_DIR, "dhair2_inform_sample.json")
+    return json.load(open(path))
+
+@pytest.fixture
+def dhair2_expected():
+    return [
+        (
+            "3.0",                      # version field
+            "2025-01-31 00:44:18+00",   # parsed from occurred
+            "patients/607",             # first item from subject array
+            "create"                    # action
+        )
+    ]
+
+@pytest.fixture
+def cosri_v2_data():
+    path = os.path.join(DATA_DIR, "cosri_v2_sample.json")
+    return json.load(open(path))
+
+@pytest.fixture
+def cosri_v2_expected():
+    return [
+        # first event: launch
+        (
+            "1",                            # version
+            "2021-09-14 17:52:31+00",       # occurred_at (zoned)
+            "Patient/1",                    # user_id
+            "launch"                        # event_type (from tags[0] or message)
+        ),
+        # second event: logout
+        (
+            "1",
+            "2021-09-17 00:04:08+00",
+            "Patient/1",
+            "logout"                        # tags[0] = "logout"
+        )
+    ]
+
+@pytest.fixture(scope="module")
+def legacy_data():
+    return json.load(open('tests/data/legacy_sample.json'))
+
+@pytest.fixture(scope="module")
+def ltt_data():
+    return json.load(open('tests/data/ltt_sample.json'))
+
+@pytest.fixture(scope="module")
+def legacy_expected():
+    return load_expected('tests/expected_results/legacy_expected.csv')
+
+@pytest.fixture(scope="module")
+def ltt_expected():
+    return load_expected('tests/expected_results/ltt_expected.csv')
```

Comment:
`load_expected()` could probably be turned into a fixture and replace these 2 file-specific fixtures, if you change `load_expected()` to accept an argument for the CSV filename